include/quic/connection.hpp

```diff
@@ -165,6 +196,30 @@ namespace oxen::quic
         }
     };
 
+	class connection_interface
+    {
```

Comment:
I don't think this should be a concrete class, but rather a pure virtual interface, along the lines of:

```C++
class connection_interface {
    public:
    	std::shared_ptr<Stream> get_new_stream(
                stream_data_callback_t data_cb = nullptr, stream_close_callback_t close_cb = nullptr) = 0;
};
```

which is then implemented by `Connection`, but we always return the connection wrapped in a `shared_ptr<connection_interface>`.

Reply:
completely agree, done and implemented